Runtime/Native/iOS/NativeClient.cs

```diff
@@ -139,35 +152,45 @@ public void HandleAnr(string gameObjectName, string callbackName)
             _anrThread = new Thread(() =>
             {
                 float lastUpdatedCache = 0;
-                while (true)
+                while (_anrThread.IsAlive && _stopAnr == false)
                 {
-                    if (lastUpdatedCache == 0)
+                    if (!_preventAnr)
                     {
-                        lastUpdatedCache = _lastUpdateTime;
-                    }
-                    else if (lastUpdatedCache == _lastUpdateTime)
-                    {
-                        if (!reported)
+                        if (lastUpdatedCache == 0)
                         {
-                            // set temporary attribute to "Hang"
-                            SetAttribute("error.type", "Hang");
-                            NativeReport("ANRException: Blocked thread detected.");
-                            // update error.type attribute in case when crash happen 
-                            SetAttribute("error.type", "Crash");
-                            reported = true;
+                            lastUpdatedCache = _lastUpdateTime;
                         }
+                        else if (lastUpdatedCache == _lastUpdateTime)
+                        {
+                            if (!reported)
+                            {
+                                // set temporary attribute to "Hang"
+                                SetAttribute("error.type", "Hang");
+                                NativeReport("ANRException: Blocked thread detected.");
+                                // update error.type attribute in case when crash happen 
+                                SetAttribute("error.type", "Crash");
+                                reported = true;
+                            }
+                        }
+                        else
+                        {
+                            reported = false;
+                        }
+
+
+                        lastUpdatedCache = _lastUpdateTime;
                     }
-                    else
+                    // make sure when ANR happened just after going to foreground

```

Comment:
These comments blocks are a bit hard to follow. I would suggest putting braces on same line or put the comment inside the block.

Reply:
done - thanks.